helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroSchemaUtil.java

```diff
@@ -529,6 +529,35 @@ public static Schema resolveUnionBranchOf(Object value, Schema union) {
     throw new IllegalStateException(sb.toString());
   }
 
+  /**
+   * checks if the value for a given schema can possibly contain
+   * strings. this is important when dealing with things like
+   * Utf8 vs java.lang.Strings
+   * @param schema a schema
+   * @return true if value under schema could possibly involve strings
+   */
+  public static boolean schemaContainsString(Schema schema) {
+    if (schema == null) {
+      return false;
+    }
+    boolean hasString = false;
+    switch (schema.getType()) {
+      case STRING:
+      case MAP: //map keys are always strings, regardless of values
+        return true;
+      case UNION:
```

Comment:
nit: we not need to iterate all union members, we can return true when seeing the first true on schemaContainsString(branch).